src/Main.java

```diff
@@ -1,5 +1,346 @@
+        int numberForOperationAdvanced;
+        String operationToPerformAdvanced;
+
+        do {
+            int previousValueOfTheSum = totalValueAdvanced;
+            operationToPerformAdvanced = enterOperation();
+
+            if (operationToPerformAdvanced.equals("+")) {
+                numberForOperationAdvanced = enterInteger();
+                totalValueAdvanced = addingIntegers(previousValueOfTheSum, numberForOperationAdvanced);
+                System.out.println("The sum of " + previousValueOfTheSum + " and " + numberForOperationAdvanced + " is " + totalValueAdvanced);
+            } else if (operationToPerformAdvanced.equals("-")) {
+                numberForOperationAdvanced = enterInteger();
+                totalValueAdvanced = subtractingIntegers(previousValueOfTheSum, numberForOperationAdvanced);
+                System.out.println("The difference of " + previousValueOfTheSum + " and " + numberForOperationAdvanced + " is " + totalValueAdvanced);
+            } else if (operationToPerformAdvanced.equals("*")) {
+                numberForOperationAdvanced = enterInteger();
+                totalValueAdvanced = multiplyingIntegers(previousValueOfTheSum, numberForOperationAdvanced);
+                System.out.println("The product of " + previousValueOfTheSum + " and " + numberForOperationAdvanced + " is " + totalValueAdvanced);
+            } else if (operationToPerformAdvanced.equals("/")) {
+                numberForOperationAdvanced = enterInteger();
+                totalValueAdvanced = divisioningIntegers(previousValueOfTheSum, numberForOperationAdvanced);
+                System.out.println("The quotient of " + previousValueOfTheSum + " and " + numberForOperationAdvanced + " is " + totalValueAdvanced);
+            }
+
+        } while (!operationToPerformAdvanced.equals("="));
+        {
+            System.out.println("Calculator result is " + totalValueAdvanced + " .");
+        }
+
+        //Task 1g
+        System.out.println("You are on task 1g \n");
+
+        int numberForOperationBonus;
+        int initialIntegerValue = 0;
+        String operationToPerformBonus;
+        int totalValueBonus = 0;
+        System.out.println("Is initial integer value defined: y/n");
+        String isIntegerValueDefined = scanner.next();
+
+        if (isIntegerValueDefined.contains("n")) {
+            System.out.println("Please enter the initial integer value:");
+            initialIntegerValue = scanner.nextInt();
+            totalValueBonus = initialIntegerValue;
+        }
+
+        // Could also be done with boolean true or false
+
+        totalValueBonus = initialIntegerValue;
+
+        do {
+            int previousValueOfTheSum = totalValueBonus;
+            System.out.println("Which operation would you like to use: *,/,+,-,=?");
+            operationToPerformBonus = scanner.next();
+
+            if (operationToPerformBonus.equals("+")) {
+                numberForOperationBonus = enterInteger();
```

Comment:
could 186 187 and 188 be a part of the same function, it would be cleaner and easier to read, and maintain

Reply:
I agree. Good point.